### PR TITLE
Introduction of close_older 

### PR DESCRIPTION
**Pull request by ruflin**
See https://github.com/elastic/filebeat/issues/181

### CODE REVIEW

File: filebeat/config/config.go [hunk 1]
```diff
@@ -15,7 +15,8 @@ import (
 // Defaults for config variables which are not set
 const (
 	DefaultRegistryFile                      = ".filebeat"
-	DefaultIgnoreOlderDuration time.Duration = 24 * time.Hour
+	DefaultIgnoreOlderDuration time.Duration = 72 * time.Hour
```
Comment (ruflin) [inline filebeat/config/config.go]: @urso I suggest to change the default to inifinite (which would be 0). WDYT? Including also @guyboertje (see https://github.com/logstash-plugins/logstash-input-file/pull/87/files#r49697378)

File: filebeat/crawler/prospector_log.go [hunk 2]
```diff
@@ -153,8 +153,8 @@ func (p ProspectorLog) checkNewFile(h *harvester.Harvester) {
 	logp.Debug("prospector", "Start harvesting unknown file: %s", h.Path)
 
 	// Check for unmodified time, but only if the file modification time is before the last scan started
-	// This ensures we don't skip genuine creations with dead times less than 10s
-	if h.Stat.Fileinfo.ModTime().Before(p.lastscan) &&
+	// If ignore_older is disable (by beeing 0) this never gets true
+	if h.Stat.Fileinfo.ModTime().Before(p.lastscan) && p.config.IgnoreOlderDuration != 0 &&
 		time.Since(h.Stat.Fileinfo.ModTime()) > p.config.IgnoreOlderDuration {
```
Comment (urso) [inline filebeat/crawler/prospector_log.go]: that's a long unreadable expression

File: filebeat/docs/configuration.asciidoc [hunk 3]
```diff
@@ -126,8 +126,26 @@ If the custom field names conflict with other field names added by Filebeat, the
...
-You can use time strings like 2h (2 hours) and 5m (5 minutes). The default is 24h.
+ignores any files that were modified before the specified timespan. This is disable by default.
+
+Files which were falling under ignore_older and are updated again, will start
+from the offset the file was at when it was last ignored by ignore_older. As an example:
+A file was not modified for 90 hours and the offset is at 200. Now a new line is added and
+the last modification date is updated. After scan_frequency detects the change the crawling
+starts at the offset 200. In case the file was falling under ignore_older already when filebeat
+was started, the first 200 lines are never sent. In case filebeat was started earlier, the 200
+chars were already sent and it now continues at the old offset.
```
Comment (urso) [inline filebeat/docs/configuration.asciidoc]: maybe move this paragraph after next one?

File: filebeat/docs/configuration.asciidoc [hunk 4]
```diff
@@ -126,8 +126,26 @@ If the custom field names conflict with other field names added by Filebeat, the
...
+
+You can use time strings like 2h (2 hours) and 5m (5 minutes). The default is 0, which means disable.
+Commenting out the config has the same affect as setting it to 0.
+
+===== close_older
+
+After a file was not modified for the duration of close_older, the file handler will be closed.
```
Comment (urso) [inline filebeat/docs/configuration.asciidoc]: s/handler/handle/

File: filebeat/docs/configuration.asciidoc [hunk 5]
```diff
@@ -126,8 +126,26 @@ If the custom field names conflict with other field names added by Filebeat, the
...
+After closing the file, a file change will only be detected after scan_frequency instead of almost
+instant.
+
+You can use time strings like 2h (2 hours) and 5m (5 minutes). The default is 1h.
```
Comment (urso) [inline filebeat/docs/configuration.asciidoc]: So, if I run out of file handles due to very fast log-rotation, I should decrease close_older only?

**Comment by urso** [comment]
LGTM

File: filebeat/docs/configuration.asciidoc (hunk 5 above)
Comment (ruflin) [inline filebeat/docs/configuration.asciidoc]: correct.

File: CHANGELOG.asciidoc [hunk 6]
```diff
@@ -21,6 +21,7 @@ https://github.com/elastic/beats/compare/1.0.0...master[Check the HEAD diff]
 - Rename proc.cpu.user_p with proc.cpu.total_p as includes CPU time spent in kernel space {pull}631[631]
 
 *Filebeat*
+- Default config for ignore_older is now inifit instead of 24h, means ignore_older is disabled by default. Use close_older to only close file handlers.
```
Comment (tsg) [inline CHANGELOG.asciidoc]: typo, s/infinit/infinite

File: filebeat/crawler/prospector_log.go [hunk 7]
```diff
@@ -153,8 +153,8 @@ func (p ProspectorLog) checkNewFile(h *harvester.Harvester) {
 	logp.Debug("prospector", "Start harvesting unknown file: %s", h.Path)
 
 	// Check for unmodified time, but only if the file modification time is before the last scan started
-	// This ensures we don't skip genuine creations with dead times less than 10s
-	if h.Stat.Fileinfo.ModTime().Before(p.lastscan) &&
+	// If ignore_older is disable (by beeing 0) this never gets true
```
Comment (tsg) [inline filebeat/crawler/prospector_log.go]: typo, s/disable/disabled

File: filebeat/docs/configuration.asciidoc [hunk 8]
```diff
@@ -126,8 +126,26 @@ If the custom field names conflict with other field names added by Filebeat, the
 ===== ignore_older
 
 If this option is specified, Filebeat
-ignores any files that were modified before the specified timespan.
-You can use time strings like 2h (2 hours) and 5m (5 minutes). The default is 24h.
+ignores any files that were modified before the specified timespan. This is disable by default.
```
Comment (tsg) [inline filebeat/docs/configuration.asciidoc]: s/disable/disabled

**Comment by tsg** [comment]
LGTM apart from the small comments above.

File: CHANGELOG.asciidoc (hunk 6 above)
Comment (ruflin) [inline CHANGELOG.asciidoc]: fixed

File: filebeat/crawler/prospector_log.go (hunk 7 above)
Comment (ruflin) [inline filebeat/crawler/prospector_log.go]: fixed

File: filebeat/docs/configuration.asciidoc (hunk 3 above)
Comment (ruflin) [inline filebeat/docs/configuration.asciidoc]: Done

File: filebeat/docs/configuration.asciidoc (hunk 4 above)
Comment (ruflin) [inline filebeat/docs/configuration.asciidoc]: fixed

File: filebeat/crawler/prospector_log.go (hunk 2 above)
Comment (ruflin) [inline filebeat/crawler/prospector_log.go]: I move it into a separate function. Still more cleanup needed (later ...)

**Comment by ruflin** [comment]
@tsg @urso have a look again.

**Pull request merged**